packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -840,6 +844,157 @@ class ThemeData with Diagnosticable {
+              textTheme: textTheme,
+            )
+            : ThemeData.from(
+              useMaterial3: useMaterial3,
+              colorScheme: colorScheme,
+              textTheme: textTheme,
+            );
+
+    if (systemColors == null) {
+      return theme;
+    }
+
+    return theme.copyWith(
+      elevatedButtonTheme: ElevatedButtonThemeData(
+        style: ElevatedButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      textButtonTheme: TextButtonThemeData(
+        style: TextButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      outlinedButtonTheme: OutlinedButtonThemeData(
+        style: OutlinedButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      filledButtonTheme: FilledButtonThemeData(
+        style: FilledButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      floatingActionButtonTheme: FloatingActionButtonThemeData(
+        backgroundColor: systemColors.buttonFace.value,
+        foregroundColor: systemColors.buttonText.value,
+      ),
+    );
+  }
+
+  /// Create a high contrast light theme based on a combination of:
+  ///
+  /// - A purple primary color that matches the baseline [Material 2](https://material.io/design/color/the-color-system.html#color-theme-creation)
+  ///   or [Material 3](https://m3.material.io/styles/color/static/baseline) color scheme.
+  /// - System colors, if available.
+  ///
+  /// This example demonstrates how to use the high contrast themes with [MaterialApp]:
+  ///
+  /// ```dart
+  /// MaterialApp(
+  ///   theme: ThemeData.light(),
+  ///   darkTheme: ThemeData.dark(),
+  ///   highContrastTheme: ThemeData.highContrastLight(),
+  ///   highContrastDarkTheme: ThemeData.highContrastDark(),
+  ///
+  ///   home: Scaffold(
+  ///     appBar: AppBar(
+  ///       title: const Text('Home'),
+  ///     ),
+  ///   ),
+  /// )
+  /// ```
```

Comment:
> Can you include a example or just a snippet of code on how user will initialize their own theme that respect the system color?

@chunhtai here's an example that uses system colors.

Reply:
oh i meant for someone that has their own theme (component theme and colorScheme), what should they do to integrate the system color?

Reply:
They should look at the implementation of `ColorScheme.highContrastLight()` 🙂

I'll try to come up with an example. Where would you recommend putting it?

Reply:
I was hoping there will be an easier migration for them without copy paste the entire code. It kind of defeat the purpose to attempt integrate systmecolor to material theme. For example, something like this
```dart
MaterialApp(
  highContrasttheme: Theme(
    textTheme: MyTextTheme(),
    colorTheme: ColorTheme.fromSeed(seed: Colors.purple),
    useSystemColor: true // this will override all color
  ),
  ...
)
```

I am also curious to hear from @QuncCccccc that how much of support we should provide in regard of the systemcolor